validator_test.sh

```diff
@@ -0,0 +1,176 @@
+fi
+echo "Chain id: $NEUTRON_CHAIN_ID"
+
+## Check if ibc connection does exist
+RES=$(neutrond query ibc connection end $CONNECTION_ID 2>/dev/null)
+
+if [ -z "$RES" ]
+then
+    echo "No such open connection for provided connection-id"
+    exit;
+fi
+echo "Connection id: $CONNECTION_ID"
+echo ""
+
+## Add a new key
+RES=$($BIN keys add $NEUTRON_KEY_NAME --output json)
+NEUTRON_ADDRESS=$(echo $RES | jq -r .address)
+MNEMONIC=$(echo $RES | jq -r .mnemonic)
+if [ $NEUTRON_ADDRESS = "null" ]
+then
+    echo "Can't get address from key"
+    exit
+fi
+
+echo "Local address in neutron: $NEUTRON_ADDRESS"
+echo "Key mnemonic: $MNEMONIC"
+echo "Key name: $NEUTRON_KEY_NAME"
+echo ""
+echo "Please go to $FAUCET_URL and get tokens for $NEUTRON_ADDRESS"
+echo "Make sure tx is passed by going to $EXPLORER_URL/accounts/$NEUTRON_ADDRESS"
+echo "Hit enter when ready"
+read
+
+## Upload contract
+echo "Upload the queries contract"
+RES=$(${BIN} tx wasm store ${CONTRACT} --from ${NEUTRON_KEY_NAME} --gas 50000000 --chain-id ${NEUTRON_CHAIN_ID} --broadcast-mode=block --gas-prices ${GAS_PRICES}  -y --output json)
+CONTRACT_CODE_ID=$(echo $RES | jq -r '.logs[0].events[1].attributes[0].value')
+if [ $CONTRACT_CODE_ID = "null" ]
+then
+    echo "Can't get code id"
+    exit
+fi
+echo "Contract code id: $CONTRACT_CODE_ID"
+echo ""
+
+## Instantiate contract
+echo "Instantiate the contract"
+INIT_CONTRACT='{}'
+RES=$(${BIN} tx wasm instantiate $CONTRACT_CODE_ID "$INIT_CONTRACT" --from $NEUTRON_KEY_NAME --admin ${NEUTRON_ADDRESS} -y --chain-id ${NEUTRON_CHAIN_ID} --output json --broadcast-mode=block --label "init"  --gas-prices ${GAS_PRICES} --gas auto --gas-adjustment 1.4)
+CONTRACT_ADDRESS=$(echo $RES | jq -r '.logs[0].events[0].attributes[0].value')
+echo "Contract address: $CONTRACT_ADDRESS"
+if [ $CONTRACT_ADDRESS = "null" ]
+then
+    echo "Can't get contract address"
+    exit
+fi
+echo ""
+
+
+## Register interchain account
+echo "Register interchain account"
+RES=$(${BIN} tx wasm execute ${CONTRACT_ADDRESS} "{\"register\": {\"connection_id\": \"${CONNECTION_ID}\", \"interchain_account_id\": \"${INTERCHAIN_ACCOUNT_ID}\"}}" --from $NEUTRON_KEY_NAME  -y --chain-id ${NEUTRON_CHAIN_ID} --output json --broadcast-mode=block --gas-prices ${GAS_PRICES} --gas 1000000)
+echo "Waiting for registering account..."
+
+## Wait until ICA appears on the target chain
+j=40
+while [[ $j -gt 0 ]]
+do
+    ((j--))
+    RES=$(neutrond query wasm contract-state smart ${CONTRACT_ADDRESS} "{\"interchain_account_address_from_contract\":{\"interchain_account_id\":\"${INTERCHAIN_ACCOUNT_ID}\"}}" --chain-id ${NEUTRON_CHAIN_ID} --output json 2>/dev/null)
+    ICA_ADDRESS=$(echo $RES | jq -r '.data | .[0]')
+    if [ ${#ICA_ADDRESS} = ${ICA_LENGTH[$i]} ]
+    then
+	break
+    fi
+    sleep 5
+done
+
+if [ ${#ICA_ADDRESS} != ${ICA_LENGTH[$i]} ]
+then
+    echo "Can't get ICA address"
+    exit
+fi
+echo "ICA address: $ICA_ADDRESS"
+echo ""
+echo "Please send 0.02 ${DENOM[$i]} to $ICA_ADDRESS"
```

Comment:
You probably wanted to write this?
```suggestion
echo "Please send 0.02 ${TARGET_DENOMS[$i]} to $ICA_ADDRESS"
```

Reply:
thanks. I wanted DENOMS bc I want to say like send 0.02 (atom/junox) and not uatom/ujunox